File: frontend/de/unika/ipd/grgen/ir/type/container/ArrayType.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 5.2
 * Copyright (C) 2003-2021 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

/**
 * @author Edgar Jakumeit
 */

package de.unika.ipd.grgen.ir.type.container;

import de.unika.ipd.grgen.ir.model.type.BaseInternalObjectType;
import de.unika.ipd.grgen.ir.type.Type;

public class ArrayType extends ContainerType
{
	public Type valueType;

	public ArrayType(Type valueType)
	{
		super("array type");
		this.valueType = valueType;
	}

	public Type getValueType()
	{
		return valueType;
	}

	@Override
	public String toString()
	{
		return "array<" + valueType + ">";
	}

	/** @see de.unika.ipd.grgen.ir.type.Type#classify() */
	@Override
	public TypeClass classify()
	{
		return TypeClass.IS_ARRAY;
	}
	
	@Override
	public Type getElementType()
	{
		return valueType;
	}

	@Override
	public boolean containsBaseInternalObjectType()
	{
		return valueType instanceof BaseInternalObjectType;
	}
}

File: frontend/de/unika/ipd/grgen/ir/type/container/ArrayType.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

/// <summary>
/// @author Edgar Jakumeit
/// </summary>

namespace de.unika.ipd.grgen.ir.type.container
{
using BaseInternalObjectType = de.unika.ipd.grgen.ir.model.type.BaseInternalObjectType;
using Type = de.unika.ipd.grgen.ir.type.Type;

public class ArrayType : ContainerType
{
	public Type valueType;

	public ArrayType(Type valueType)
		: base("array type")
	{
		this.valueType = valueType;
	}

	public virtual Type ValueType
	{
		get
		{
		return valueType;
		}
	}

	public override string ToString()
	{
		return "array<" + valueType + ">";
	}

	/// <seealso cref="de.unika.ipd.grgen.ir.type.Type.classify() "/>
	public override TypeClass Classify()
	{
		return TypeClass.IS_ARRAY;
	}

	public override Type ElementType
	{
		get
		{
		return valueType;
		}
	}

	public override bool ContainsBaseInternalObjectType()
	{
		return valueType is BaseInternalObjectType;
	}
}

}
